tests-arquillian/src/test/java/org/jboss/weld/tests/event/lifecycle/ordering/PoorWorker.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.tests.event.lifecycle.ordering;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+/**
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+@ApplicationScoped
+public class PoorWorker {
+
+    @Inject
+    MassiveJugCoffee coffee;
+
+    @Produces
+    public HighQualityAndLowCostProduct makeManagersHappy(String docs) {
+        return new HighQualityAndLowCostProduct();
+    }
+
+    @Produces
```

Comment:
It's a good idea to use a qualifier for any `String` bean type. Otherwise, we risk ambiguous dependencies. E.g. we had similar problems with beans produced by Batch properties before `@BatchProperty` was introduced.